### DISCUSSION TITLE
**/.env published to npm by default.

### DISCUSSION THREAD

**Discussion by showierdata9978** [pnpm/pnpm#16001] [Ideas]
### Verify latest release

- [X] I verified that the issue exists in the latest pnpm release

### pnpm version

8.15.5

### Which area(s) of pnpm are affected? (leave empty if unsure)

CLI

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/meower-media-co/meower.js

### Reproduction steps

- create a package.
- create a .env file
- publish package

### Describe the Bug

.env files are included by default, but this poses a major security risk, due to leaking confidential passwords, when pnpm starts to publish the entire workspace instead of the files you intend to (ie: dist, browser).

### Expected Behavior

.env files are NOT included by default.

### Which Node.js version are you using?

v20.10.0

### Which operating systems have you used?

- [ ] macOS
- [ ] Windows
- [X] Linux

### If your OS is a Linux based, which one it is? (Include the version if relevant)

Fedora

**Comment by MisterJimson** [comment]
I’m working on a fix for this.

---
Written by an agent (Codex, gpt-6-astra).
